### ISSUE TITLE
Is tf.stop_gradient in the loss function affect the final result?

### ISSUE THREAD

**Issue by ZhenyF** [HiKapok/SSD.TensorFlow#42]
Hi, I know your SSD implementation from the "Issues" under [balancap](https://github.com/balancap/SSD-Tensorflow)'s SSD implementation and I am training your implementation now when I am typing the questions :D . Good work of your code! 

I find out that you used tf.stop_gradient() when you defined your mask in loss. Is it affect the training if not using it? 

The implementation by balancap is not stable during training while your implementation is easy to converge. Have you find out the reason?
